receiver/splunkenterprisereceiver/scraper.go

```diff
@@ -193,3 +201,430 @@ func (s *splunkScraper) scrapeIndexThroughput(ctx context.Context, now pcommon.T
 		s.mb.RecordSplunkIndexerThroughputDataPoint(now, 1000*entry.Content.AvgKb, entry.Content.Status)
 	}
 }
+
+// Scrape indexes extended total size
+func (s *splunkScraper) scrapeIndexesTotalSize(ctx context.Context, now pcommon.Timestamp, errs *scrapererror.ScrapeErrors) {
+	var it IndexesExtended
+	var ept string
+
+	if !s.conf.MetricsBuilderConfig.Metrics.SplunkDataIndexesExtendedTotalSize.Enabled {
+		return
+	}
+
+	ept = apiDict[`SplunkDataIndexesExtended`]
+
+	req, err := s.splunkClient.createAPIRequest(ctx, ept)
+	if err != nil {
+		errs.Add(err)
+	}
+
+	res, err := s.splunkClient.makeRequest(req)
+	if err != nil {
+		errs.Add(err)
+		return
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		errs.Add(err)
+	}
+
+	err = json.Unmarshal(body, &it)
+	if err != nil {
+		errs.Add(err)
+	}
+
+	var name string
+	var totalSize int64
+	for _, f := range it.Entries {
+		if f.Name != "" {
+			name = f.Name
+		}
+		if f.Content.TotalSize != "" {
+			mb, err := strconv.ParseFloat(f.Content.TotalSize, 64)
+			totalSize = int64(mb * 1024 * 1024)
+			if err != nil {
+				fmt.Println("Failed to parse total_size from response")
```

Comment:
I think instead of printing to stdout we may want to properly log these messages. Perhaps a `logger` should be included with the `Scraper` object. This applies to all instances of fmt.Println below.